ui/tooltip.js

```diff
@@ -280,14 +280,28 @@ $.widget( "ui.tooltip", {
 		tooltipData = this._tooltip( target );
 		tooltip = tooltipData.tooltip;
 		this._addDescribedBy( target, tooltip.attr( "id" ) );
+		// Parse content into DOM remove attributes id and name
+		if ( typeof content === "string" || content.nodeType ) {
+			if ( typeof content === "string" ) {
+				tooltip_content_div = document.createElement("div");
+				tooltip_content_div.innerHTML = content;
+				dom_content = $(tooltip_content_div);
+			} else {
+				dom_content = $("<div>").html( content );
+			}
+			dom_content.find( "[id]" ).removeAttr( "id" );
+			dom_content.find( "[name]" ).removeAttr( "name" );
+			content = dom_content.html();
+		} else if ( content.jquery ) {
+			content.removeAttr( "id" ).find( "[id]" ).removeAttr( "id" );
+			content.removeAttr( "name" ).find( "[name]" ).removeAttr( "name" );
+		}
 		tooltip.find( ".ui-tooltip-content" ).html( content );
```

Comment:
I would say 
Firstly essentially it is bad practice to do this and the user could definitely find a way around this problem them self. 
Secondly the reason I wanted to fix this issue other than for the user who reported it was that there are only a few things you can put in a tooltip that are going to have external effects. name and id do cause those effects, why not ensure they don't? 
Finally I agree if you think this may not be a good fix as its not immediately apparent why a users id would disappear if they were attempting to select inside the tooltip. In which case it would be fixing one niche use-case and breaking another.